server/src/main/java/org/elasticsearch/action/admin/indices/template/put/PutIndexTemplateRequest.java

```diff
@@ -539,4 +540,34 @@ public void writeTo(StreamOutput out) throws IOException {
         }
         out.writeOptionalVInt(version);
     }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        if (customs.isEmpty() == false) {
+            throw new IllegalArgumentException("Custom data type is no longer supported in index template [" + customs + "]");
+        }
+        builder.field("index_patterns", indexPatterns);
+        builder.field("order", order);
+        if (version != null) {
+            builder.field("version", version);
+        }
+
+        builder.startObject("settings");
+        settings.toXContent(builder, params);
+        builder.endObject();
+
+        builder.startObject("mappings");
+        for (Map.Entry<String, String> entry : mappings.entrySet()) {
+            Map<String, Object> mapping = XContentHelper.convertToMap(new BytesArray(entry.getValue()), false).v2();
+            builder.field(entry.getKey(), mapping);
```

Comment:
could of things here: we should not use a deprecated method, I think it is guaranteed that the request holds json (we convert it on the setter methods), hence we should not need content-type auto-detection. Furthermore, we don't need to parse stuff back into a map to the write it back to the builder. I would try and do something like this:

```
builder.field(entry.getKey());
XContentParser parser = JsonXContent.jsonXContent.createParser(NamedXContentRegistry.EMPTY, DeprecationHandler
                    .THROW_UNSUPPORTED_OPERATION, entry.getValue());
builder.copyCurrentStructure(parser);
```